ocaml/xapi/repository_helpers.ml

```diff
@@ -173,12 +173,17 @@ module Guidance = struct
         let msg = error_msg l in
         raise Api_errors.(Server_error (internal_error, [msg]))
 
-  let resort_guidances gs =
+  let resort_guidances ~kind gs =
     match GuidanceStrSet.find_opt "RebootHost" gs with
```

Comment:
I think there are some unnecessary conversions between strings and `Guidance.t` (see `Guidance.eq` for example). We could use a set of `Guidance.t` instead (and we get some more type safety)

```
module GuidanceSet = Set.Make(struct type t = Guidance.t ;; let compare = Stdlib.compare ) ;;

..

match GuidanceSet.find_opt RebootHost gs with
| Some _ -> GuidanceSet.singleton RebootHost
...
```